app/Filament/App/Resources/ServerResource/Pages/ListServers.php

```diff
@@ -73,4 +143,40 @@ public function getTabs(): array
                 ->badge($all->count()),
         ];
     }
+
+    public function getResourceColor(Server $server, string $resource): ?string
+    {
+        $current = null;
+        $limit = null;
+
+        if ($resource === 'cpu') {
+            $current = $server->resources()['cpu_absolute'] ?? 0;
+            $limit = $server->cpu;
+
+            if ($server->cpu === 0) {
+                return null;
+            }
+        } elseif ($resource === 'memory') {
+            $current = $server->resources()['memory_bytes'] ?? 0;
+            $limit = $server->memory * 2 ** 20;
+            if ($server->memory === 0) {
+                return null;
+            }
+
+        } elseif ($resource === 'disk') {
+            $current = $server->resources()['disk_bytes'] ?? 0;
+            $limit = $server->disk * 2 ** 20;
+            if ($server->disk === 0) {
+                return null;
+            }
+        }
```

Comment:
Can you make this into a switch statement instead.

Reply:
Sure i can `switch` it up a little :P